File: Interpreted/lcs-recursive/lcs.pl
sub lcs {
    my ($xstr, $ystr) = @_;
    ($xstr eq '' or $ystr eq '') && return '';

    my ($x, $xs, $y, $ys) = (substr($xstr, 0, 1), substr($xstr, 1),
                             substr($ystr, 0, 1), substr($ystr, 1));

    if ($x eq $y) {
        return ($x . lcs($xs, $ys));
    }

    my $lcs1 = lcs($xstr, $ys);
    my $lcs2 = lcs($xs,   $ystr);
    length($lcs1) > length($lcs2) ? $lcs1 : $lcs2;
}

print lcs($ARGV[0], $ARGV[1]), "\n";

File: Interpreted/lcs-recursive/lcs.php
<?php
function lcs($xstr, $ystr) {

    if ($xstr == '' or $ystr == '') {
        return '';
    }

    $x = $xstr[0];
    $xs = substr($xstr, 1);
    $y = $ystr[0];
    $ys = substr($ystr, 1);

    if ($x == $y) {
        return ($x . lcs($xs, $ys));
    }

    $lcs1 = lcs($xstr, $ys);
    $lcs2 = lcs($xs, $ystr);

    return strlen($lcs1) > strlen($lcs2) ? $lcs1 : $lcs2;
}

echo lcs($argv[1], $argv[2]), "\n";
?>
